apache2/modsecurity.c

```diff
@@ -241,40 +193,32 @@ int modsecurity_init(msc_engine *msce, apr_pool_t *mp) {
 /**
  * Performs per-child (new process) initialisation.
  */
-void modsecurity_child_init(msc_engine *msce) {
+void modsecurity_child_init(apr_pool_t *mp, msc_engine *msce) {
     /* Need to call this once per process before any other XML calls. */
     xmlInitParser();
 
-    if (msce->auditlog_lock != NULL) {
-        apr_status_t rc = apr_global_mutex_child_init(&msce->auditlog_lock, NULL, msce->mp);
-        if (rc != APR_SUCCESS) {
-            // ap_log_error(APLOG_MARK, APLOG_ERR, rs, s, "Failed to child-init auditlog mutex");
-        }
+    if (msce->auditlog_lock == NULL) {
+        msce->auditlog_lock = apr_pcalloc(mp, sizeof(struct waf_lock));
     }
+    Waf_createLock(msce->auditlog_lock, "auditlog_lock", strlen("auditlog_lock"), msc_waf_lock_owner);
```

Comment:
Is this idempotent? It just reuses the lock if it already exists?

Reply:
yes, if lock exists, it will reuse